lib/segment/src/spaces/simple.rs

```diff
@@ -201,6 +221,28 @@ pub fn dot_similarity(v1: &[VectorElementType], v2: &[VectorElementType]) -> Sco
     v1.iter().zip(v2).map(|(a, b)| a * b).sum()
 }
 
+pub fn jaccard_similarity(v1: &[VectorElementType], v2: &[VectorElementType]) -> ScoreType {
+    // Assumes binary vectors
+
+    if v1.is_empty() || v2.is_empty() {
+        return 1.0;
+    }
+
+    let interserction_len: i64 = v1
+        .iter()
+        .zip(v2.iter())
+        .map(|(a, b)| (*a as i64) & (*b as i64))
```

Comment:
Does this implementation assumes that float vectors will be interpreted as binary sets?

What do you think about continuous interpretation of Jaccard distance? E.g. https://en.wikipedia.org/wiki/Jaccard_index#Weighted_Jaccard_similarity_and_distance

Should be simple enough to implement and also naturally flows into the usual jaccard distnace, once we implement binary vectors support

Reply:
@sandkoan in https://github.com/qdrant/qdrant/issues/1319 asked for binary sets, however, I agree that it's awkward to do all the conversions from float to binary and back. Sound like it's better to have binary vector support first.

I can look into changing this to the float vector version.